fix readme badge and revise readme (#276)
fkie-cad/Logprep: README.md

Logprep allows to collect, process and forward log messages from various data sources.
Log messages are being read and written by so-called connectors.
Currently, connectors for Kafka, Elasticsearch, Opensearch and JSON(L) files exist.

> Additionally an Input Connector for HTTP Input is provided, which starts an uvicorn server and

Reviewer: ```suggestion
Additionally, an Input Connector for HTTP Input is provided, which starts an uvicorn server and
```

Logprep processes incoming log messages with a configured pipeline that can be spawned
multiple times via multiprocessing.
The following chart shows a basic setup that represents this behaviour.

> The pipeline consists of three processors the `Dissector`, `Geo-IP Enricher` and the 

Reviewer: ```suggestion
The pipeline consists of three processors: the `Dissector`, `Geo-IP Enricher` and the 
```

### Processors

Every processor has one simple task to fulfill.

> For example, the `Dissector` can split up long message fields into multiple subfields
> leading to a normalized message pattern.

Reviewer: Does the Dissector really fulfil every requirement for normalization?

```suggestion
For example, the `Dissector` can split up long message fields into multiple subfields
to facilitate structural normalization.
```

Connectors are responsible for reading the input and writing the result to a desired output. 
The main connectors that are currently used and implemented are a kafka-input-connector and a
kafka-output-connector allowing to receive messages from a kafka-topic and write messages into a

> kafka-topic. Addionally you can use the Opensearch or Elasticsearch output connectors to ship the

Reviewer: ```suggestion
kafka-topic. Addionally, you can use the Opensearch or Elasticsearch output connectors to ship the
```

### Configuration

> To run Logprep, certain configurations have to be provided. Because Logprep is designed to run in a
> containerized environment like Kubernetes these configurations can be provided via filesystem or
> http. By providing the configuration via http it is possible to control the configuration change via
> a flexible http api to make Logprep fast reacting to changes in your environment.
> 

Reviewer: ```suggestion
To run Logprep, certain configurations have to be provided. Because Logprep is designed to run in a
containerized environment like Kubernetes, these configurations can be provided via the filesystem or
http. By providing the configuration via http, it is possible to control the configuration change via
a flexible http api. This enables Logprep to quickly adapt to changes in your environment.

```

Instead, it can be issued to reload the configuration. 
For this, the signal `SIGUSR1` must be send to the Logprep process.

> Besides this a `config_refresh_interval` can be configured to make a automatic config refresh.

Reviewer: ```suggestion
Additionally, a `config_refresh_interval` can be set to periodically and automatically refresh the given configuration.
```

> This can be useful in containerizes environments like kubernetes when mounted pod volumes changes
> on the fly.

Reviewer: ```suggestion
This can be useful in case of containerized environments (such as Kubernetes), when pod volumes often change
on the fly.
```
